sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -182,309 +182,61 @@ private static String namespaceOf(ClassForDisplay clazz) {
-     */
-    ItemBuilder withLinkUrl(@Nullable String url);
+  @AutoValue
+  public abstract static class Item<T> {
 
-    /**
-     * Adds an explicit namespace to the most-recently added display data. The namespace
-     * and key uniquely identify the display data.
-     *
-     * <p>Leaving the namespace unspecified will default to the registering instance's class.
-     */
-    ItemBuilder withNamespace(Class<?> namespace);
-
-    /**
-     * Adds an explicit namespace to the most-recently added display data. The namespace
-     * and key uniquely identify the display data.
-     *
-     * <p>Leaving the namespace unspecified will default to the registering instance's class.
-     */
-    ItemBuilder withNamespace(ClassForDisplay namespace);
-  }
-
-  /**
-   * A display data item. DisplayData items are registered via {@link Builder#add} within
-   * {@link HasDisplayData#populateDisplayData} implementations. Each metadata item is uniquely
-   * identified by the specified key and namespace generated from the registering component's
-   * class name.
-   */
-  public static class Item {
-    private final String key;
-    private final String ns;
-    private final Type type;
-    private final Object value;
-    private final Object shortValue;
-    private final String label;
-    private final String url;
-
-    private static Item create(String nsClass, String key, Type type, Object value) {
-      FormattedItemValue formatted = type.format(value);
-      return new Item(
-          nsClass, key, type, formatted.getLongValue(), formatted.getShortValue(), null, null);
-    }
-
-    private Item(
-        String namespace,
-        String key,
-        Type type,
-        Object value,
-        Object shortValue,
-        String url,
-        String label) {
-      this.ns = namespace;
-      this.key = key;
-      this.type = type;
-      this.value = value;
-      this.shortValue = shortValue;
-      this.url = url;
-      this.label = label;
-    }
-
-    @JsonGetter("namespace")
-    public String getNamespace() {
-      return ns;
-    }
+    @Nullable
+    public abstract String getNamespace();
 
     @JsonGetter("key")
-    public String getKey() {
-      return key;
-    }
+    public abstract String getKey();
 
     /**
      * Retrieve the {@link DisplayData.Type} of display data. All metadata conforms to a
      * predefined set of allowed types.
      */
     @JsonGetter("type")
-    public Type getType() {
-      return type;
-    }
+    public abstract Type getType();
 
     /**
      * Retrieve the value of the metadata item.
      */
     @JsonGetter("value")
-    public Object getValue() {
-      return value;
-    }
+    @Nullable
+    public abstract Object getValue();
 
     /**
      * Return the optional short value for an item. Types may provide a short-value to displayed
-     * instead of or in addition to the full {@link Item#value}.
+     * instead of or in addition to the full {@link Item#getValue() value}.
```

Comment:
Is this always a string or can the type change? Must it be the same type?

Maybe update the Javadoc to reflect the fact that the value is always set, the short value is optional, and whether it is actually displayed depends on the consumer of display data.